src/ibrowse_http_client.erl

```diff
@@ -646,6 +646,11 @@ filter_sock_options(Opts) ->
                          true
                  end, Opts).
 
+%% If we don't have a body, we avoid the send as the other side may
+%% have already closed the connection in the case of non-keepalive
+%% connections.
+do_send([], _)  ->  ok;
+do_send(<<>>, _)  ->  ok;
```

Comment:
do_send is used to send the headers, etc as well as the simple body, and so it's a little bit confusing even if it the code is correct to refer to the body here. It would seem clearer to test this as part of do_send_body below. Am I missing something here?

Reply:
🤔 Definitely semantically I am +1 on moving the check. Although, this has me wondering...is there ever a point in calling this function with nothing to send?

Reply:
Probably not, unless there is something strange going down in the TCP or SSL stacks that do something with empty packets.